src/helpers/__tests__/validationHelpers.spec.js

```diff
@@ -0,0 +1,29 @@
+import { describe, expect, it } from "vitest";
+import { hasOneDigit, isLongEnough, isUserOlderThan } from "@/helpers/validationHelpers";
+
+describe("hasOneDigit", () => {
+  it("returns true when a given string includes at least one digit", () => {
+    expect(hasOneDigit("Monte5")).toEqual(true);
+  });
+  it("returns false when a given string doesn't include any digit", () => {
+    expect(hasOneDigit("MonteMonte!")).toEqual(false)
+  })
+})
+
+describe("isLongEnough", () => {
+  it("returns true when a given string has at least a given number of characters", () => {
+    expect(isLongEnough("Monterail", 8)).toEqual(true)
+  });
+  it("returns false when a given string doesn't have at a given minimum number of characters", () => {
+    expect(isLongEnough("Monte", 8)).toEqual(false)
+  });
+})
+
+describe("isUserOlderThan", () => {
+  it("returns true if user born on a given day is of/older than a given age", () => {
+    expect(isUserOlderThan("2000-01-01", 21)).toEqual(true);
+  });
+  it("returns false if user born on a given day is younger than a given age", () => {
+    expect(isUserOlderThan("2010-01-01", 21)).toEqual(false)
+  })
+})
```

Comment:
this is tricky but this test will start failing in 9 years from now 🙈  we are trying to write tests in such a way that it will always pass/fail if tested method has not been changed.

https://vitest.dev/guide/mocking.html#dates

Reply:
Wow, good point! 💪 I think I have managed to fix that (although without fake-timer).